### PR TITLE
[KOGITO-4700] BDD tests for Data Index Service with PostgreSQL persis…

### PR DESCRIPTION
**Pull request by sutaakar**
Signed-off-by: Karel Suta <ksuta@redhat.com>

https://issues.redhat.com/browse/KOGITO-4700
https://issues.redhat.com/browse/KOGITO-4698

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/main/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've ran `make before-pr` and everything is working accordingly
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](https://github.com/kiegroup/kogito-operator/blob/main/RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

**Comment by sutaakar** [comment]
Marked as WIP as the scenarios don't work properly yet due to many PostgreSQL issues.

**Comment by sutaakar** [comment]
Implementing together with https://issues.redhat.com/browse/KOGITO-4698 as the scenarios are interconnected.

File: test/features/install_dataindex.feature [hunk 1]
```diff
@@ -62,4 +62,31 @@ Feature: Kogito Data Index
...
+  Scenario: Install Kogito Data Index with PostgreSQL
+    Given PostgreSQL instance "postgresql" is deployed within 3 minutes with configuration:
+      | username | myuser |
+      | password | mypass |
+      | database | mydb   |
+
+    When Install Kogito Data Index with 1 replicas with configuration:
+      | config | database-type | PostgreSQL               |
+      | config | infra         | kafka                    |
+      | runtime-env | QUARKUS_DATASOURCE_JDBC_URL     | jdbc:postgresql://postgresql:5432/mydb |
```
Comment (radtriste) [inline test/features/install_dataindex.feature:77]: using both URLs here but only one in scenario https://github.com/kiegroup/kogito-operator/pull/950/files#diff-4e0f459bc0fdab9e8a6bf7bce0c0c4b320fc5dfd7f751b4cc4e436c4286a7df0R433 ?

File: test/pkg/steps/kogitodataindex.go [hunk 2]
```diff
@@ -55,6 +55,9 @@ func (data *Data) installKogitoDataIndexServiceWithReplicasWithConfiguration(rep
 	if dataIndex.DatabaseType == infrastructure.MongoDBKind {
 		framework.GetMainLogger().Debug("Setting Data Index MongoDB image")
 		dataIndex.KogitoService.GetSpec().SetImage(framework.NewImageOrDefault(config.GetDataIndexImageTag(), kogitosupportingservice.DataIndexMongoDBImageName))
+	} else if dataIndex.DatabaseType == "PostgreSQL" {
+		framework.GetMainLogger().Debug("Setting Data Index PostgreSQL image")
+		dataIndex.KogitoService.GetSpec().SetImage(framework.NewImageOrDefault(config.GetJobsServiceImageTag(), kogitosupportingservice.DataIndexPostgresqlImageName))
```
Comment (radtriste) [inline test/pkg/steps/kogitodataindex.go:60]:
```suggestion
		dataIndex.KogitoService.GetSpec().SetImage(framework.NewImageOrDefault(config.GetDataIndexImageTag(), kogitosupportingservice.DataIndexPostgresqlImageName))
```

**Comment by sutaakar** [comment]
Waiting for https://github.com/kiegroup/kogito-examples/pull/815

File: test/features/deploy_kogito_runtime.feature [hunk 3]
```diff
@@ -408,6 +475,80 @@ Feature: Deploy Kogito Runtime
...
+      | profile | events,persistence |
+      | native  | <native>           |
+
+    When Deploy <runtime> example service "<example-service>" from runtime registry with configuration:
+      | config | infra | kafka            |
+      # Setup short name as it can create some problems with route name too long ...
+      | config      | name                                         | process-postgres                                |
+      # For Quarkus
+      | runtime-env | quarkus.datasource.username                  | myuser                                          |
+      | runtime-env | quarkus.datasource.password                  | mypass                                          |
```
Comment (radtriste) [inline test/features/deploy_kogito_runtime.feature:510]: why not using `QUARKUS_DATASOURCE_USERNAME` and `QUARKUS_DATASOURCE_PASSWORD` ?

**Review by radtriste — APPROVED** [review]
one minor comment. otherwise lgtm

File: test/features/deploy_kogito_runtime.feature (hunk 3 above)
Comment (sutaakar) [reply test/features/deploy_kogito_runtime.feature]: Replaced all env variables with uppercase format

**Comment by radtriste** [comment]
Failing test is Grafana one

**Review by MarianMacik — APPROVED** [review]
Looks good to me.

**Pull request merged**